google-cloud-trace/lib/google/cloud/trace.rb

```diff
@@ -50,31 +50,100 @@ module Cloud
-    # your web browser and navigate to the "Trace" section. Your application
-    # latency traces will appear there; no additional coding is required.
-    #
-    # You may also adopt Stackdriver diagnostic services "a la carte" by
-    # installing individual gems. To instrument your app for the Trace service,
-    # add this gem, `google-cloud-trace`, to your Gemfile. Then add the
-    # following line to your `config/application.rb` file:
+    # This library integrates with Rack-based web frameworks such as Ruby On
+    # Rails to provide latency trace reports for your application.
+    # Specifcally, it:
+    #
+    # *   Provides a Rack middleware that automatically reports latency traces
+    #     for http requests handled by your application, and measures the
+    #     latency of each request as a whole.
+    # *   Integrates with `ActiveSupport::Notifications` to add important
+    #     latency-affecting events such as ActiveRecord queries to the trace.
+    # *   Provides a simple API for your application code to define and
+    #     measure latency-affecting processes specific to your application.
+    #
+    # When this library is installed and configured in your running
+    # application, you can view your application's latency traces in real time
+    # by opening the Google Cloud Console in your web browser and navigating
+    # to the "Trace" section. It also integrates with Google App Engine
+    # Flexible and Google Container Engine to provide additional information
+    # for applications hosted in those environments.
+    #
+    # ### Using instrumentation with Ruby on Rails
+    #
+    # To install application instrumentation in your Ruby on Rails app, add
+    # this gem, `google-cloud-trace`, to your Gemfile and update your bundle.
+    # Then add the following line to your `config/application.rb` file:
     #
     # ```ruby
     # require "google/cloud/trace/rails"
     # ```
     #
+    # This will install a Railtie that automatically integrates with the
+    # Rails framework, installing the middleware and the ActiveSupport
+    # integration for you. Your application traces, including basic request
+    # tracing, ActiveRecord query measurements, and view render measurements,
+    # should then start appearing in the Cloud Console.
+    #
     # See the {Google::Cloud::Trace::Railtie} class for more information,
     # including how to customize your application traces.
     #
-    # If your application uses a different Rack-based web framework, you may
-    # instrument your app by installing the trace middleware. See the
-    # {Google::Cloud::Trace::Middleware} documentation for more information.
+    # ### Using instrumentation with Sinatra
+    #
+    # To install application instrumentation in your Sinatra app, add this gem,
+    # `google-cloud-trace`, to your Gemfile and update your bundle. Then add
+    # the following lines to your main application Ruby file:
+    #
+    # ```ruby
```

Comment:
Unfortunately, the `@example` tag groups all examples at the end of a document block. In these docs, I want the examples inline.